libs/partners/ai21/tests/unit_tests/test_semantic_text_splitter.py

```diff
@@ -0,0 +1,126 @@
+from unittest.mock import Mock
+
+import pytest
+
+from langchain_ai21 import AI21SemanticTextSplitter
+
+TEXT = (
+    "The original full name of the franchise is Pocket Monsters (ポケットモンスター, "
+    "Poketto Monsutā), which was abbreviated to "
+    "Pokemon during development of the original games.\n"
+    "When the franchise was released internationally, the short form of the title was "
+    "used, with an acute accent (´) "
+    "over the e to aid in pronunciation.\n"
+    "Pokémon refers to both the franchise itself and the creatures within its "
+    "fictional universe.\n"
+    "As a noun, it is identical in both the singular and plural, as is every "
+    "individual species name;[10] it is "
+    'grammatically correct to say "one Pokémon" and "many Pokémon", as well '
+    'as "one Pikachu" and "many Pikachu".\n'
+    "In English, Pokémon may be pronounced either /'powkɛmon/ (poe-keh-mon) or "
+    "/'powkɪmon/ (poe-key-mon).\n"
+    "The Pokémon franchise is set in a world in which humans coexist with creatures "
+    "known as Pokémon.\n"
+    "Pokémon Red and Blue contain 151 Pokémon species, with new ones being introduced "
+    "in subsequent games; as of December 2023, 1,025 Pokémon species have been "
+    "introduced.\n[b] Most Pokémon are inspired by real-world animals;[12] for example,"
+    "Pikachu are a yellow mouse-like species[13] with lightning bolt-shaped tails[14] "
+    "that possess electrical abilities.[15]"
+)
+
+
+@pytest.mark.parametrize(
+    ids=[
+        "when_chunk_size_is_zero",
+        "when_chunk_size_is_large",
+        "when_chunk_size_is_small",
+    ],
+    argnames=["chunk_size"],
+    argvalues=[
+        (0,),
+        (1000,),
+        (10,),
+    ],
+)
+def test_split_text__on_chunk_size(
+    chunk_size: int,
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts = AI21SemanticTextSplitter(
+        chunk_size=chunk_size,
+        client=mock_client_with_semantic_text_splitter,
+    )
+    response = sts.split_text("This is a test")
+    assert len(response) > 0
```

Comment:
Can you maybe test a more precise response len here that correlates to each chunk size?